src/components/hand-controls.js

```diff
@@ -74,34 +126,95 @@ module.exports.Component = registerComponent('hand-controls', {
    * @param {string} evt the event associated to the button
    */
   handleButton: function (button, evt) {
-    var el = this.el;
     var isPressed = evt === 'down';
-    switch (button) {
-      case 'trackpad':
-        if (isPressed === this.trackpadPressed) { return; }
-        this.trackpadPressed = isPressed;
-        this.playAnimation('thumb', !isPressed);
-        evt = isPressed ? 'thumbup' : 'thumbdown';
-        el.emit(evt);
-        break;
-      case 'trigger':
-        if (isPressed === this.triggerPressed) { return; }
-        this.triggerPressed = isPressed;
-        this.playAnimation('pointing', !isPressed);
-        evt = isPressed ? 'pointup' : 'pointdown';
-        el.emit(evt);
-        break;
-      case 'grip':
-        if (isPressed === this.gripPressed) { return; }
-        this.gripPressed = isPressed;
-        this.playAnimation('close', !isPressed);
-        evt = isPressed ? 'gripclose' : 'gripopen';
-        el.emit(evt);
-        break;
+    var isTouched = evt === 'touchstart';
+    if (evt.indexOf('touch') === 0) {
+      if (isTouched === this.touchedButtons[button]) { return; }
+      this.touchedButtons[button] = isTouched;
+    } else {
+      if (isPressed === this.pressedButtons[button]) { return; }
+      this.pressedButtons[button] = isPressed;
+    }
+    if (!this.determineGesture()) { return; }
```

Comment:
It's a a bit cryptic what this function returns. Let's handle the update here:
````javascript
var gesture = this.getGesture();
if (this.gesture === gesture) { return; }
this.gesture = gesture;
this.animageGesture(gesture);
````